Makefile

```diff
@@ -57,21 +73,17 @@ build-natives:
 	@env GOOS=windows GOARCH=$$arch CC=x86_64-w64-mingw32-gcc CXX=x86_64-w64-mingw32-g++ \
 		$(GO) build $(GO_LDFLAGS_WIN) $(BUILDV) -tags $(TAGS) nolzma -o $(PKGNAME)-windows.exe ;
 
-build-contained:
-	rm -f mender-artifact && \
-	image_id=$$(docker build -f Dockerfile . | awk '/Successfully built/{print $$NF;}') && \
-	docker run --rm --entrypoint "/bin/sh" -v $(shell pwd):/binary $$image_id -c "cp /go/bin/mender-artifact /binary" && \
-	docker image rm $$image_id
+build-contained: $(DOCKER_BUILD_FLAG_BASE)
```

Comment:
This doesn't take into account that the source might have changed. I guess it's fixable, but checking for all the source file will complicate it a bit. I guess the question is whether it's worth the hassle.

Reply:
Yes, you're right, at the moment it only looks for changes in the related Dockerfile.
I'm still looking into making it aware of Go source file changes too but, as you said, it probably won't be easy as Go projects structure is not really Makefile-friendly. If you deem this as overcomplicated/overengineered, I can leave it as that by reverting the last commit.

Reply:
Yes, given that the benefit is pretty low (Go already caches build artifacts), I think I would vote for omitting this to keep the complexity down.

Reply:
This commit should be removed, as per the discussion.

There is definitely room for improvement here, but we can't submit something that doesn't take code changes into account.